spark/src/main/scala/geotrellis/spark/clip/ClipToGrid.scala

```diff
@@ -16,191 +16,111 @@
-      k: SpatialKey,
-      preds: Predicates
-    ): Option[(SpatialKey, Feature[Geometry, D])] = {
-      val extent = mapTransform(k)
-
-      clipFeature(extent, feature, preds).map(k -> _)
+  private def coveredBy[G <: Geometry](g: G, e: Extent): Boolean =
+    g.jtsGeom.coveredBy(e.toPolygon.jtsGeom)
+
+  private def clip[G <: Geometry](e: Extent, g: G): Option[Geometry] = {
+    val exPoly: Polygon = e.toPolygon
+
+    val clipped: Try[Geometry] = g match {
+      case mp: MultiPolygon => Try(MultiPolygon(mp.polygons.flatMap(_.intersection(exPoly).as[Polygon])))
+      case _ => Try(g.intersection(exPoly).toGeometry.get)
     }
 
-    val iterator: Iterator[(SpatialKey, Feature[Geometry, D])] =
-      feature.geom match {
-        case p: Point =>
-          val k = mapTransform(p)
-          clipToKey(k, pointPredicates).toSeq.iterator
-        case mp: MultiPoint =>
-          mp.points
-            .map(mapTransform(_))
-            .distinct
-            .map(clipToKey(_, mpOrLinePredicates))
-            .flatten
-            .iterator
-        case l: Line =>
-          mapTransform.multiLineToKeys(MultiLine(l))
-            .map(clipToKey(_, mpOrLinePredicates))
-            .flatten
-        case ml: MultiLine =>
-          mapTransform.multiLineToKeys(ml)
-            .map(clipToKey(_, mpOrLinePredicates))
-            .flatten
-        case p: Polygon =>
-          val pg = PreparedGeometryFactory.prepare(p.jtsGeom)
-          val preds = polyPredicates(pg)
-
-          mapTransform
-            .multiPolygonToKeys(MultiPolygon(p))
-            .map(clipToKey(_, preds))
-            .flatten
-        case mp: MultiPolygon =>
-          val pg = PreparedGeometryFactory.prepare(mp.jtsGeom)
-          val preds = polyPredicates(pg)
-
-          mapTransform.multiPolygonToKeys(mp)
-            .map(clipToKey(_, preds))
-            .flatten
-        case gc: GeometryCollection =>
-          def keysFromGC(g: GeometryCollection): List[SpatialKey] = {
-            var keys: List[SpatialKey] = List()
-            keys = keys ++ gc.points.map(mapTransform.apply)
-            keys = keys ++ gc.multiPoints.flatMap(_.points.map(mapTransform.apply))
-            keys = keys ++ gc.lines.flatMap { l => mapTransform.multiLineToKeys(MultiLine(l)) }
-            keys = keys ++ gc.multiLines.flatMap { ml => mapTransform.multiLineToKeys(ml) }
-            keys = keys ++ gc.polygons.flatMap { p => mapTransform.multiPolygonToKeys(MultiPolygon(p)) }
-            keys = keys ++ gc.multiPolygons.flatMap { mp => mapTransform.multiPolygonToKeys(mp) }
-            keys = keys ++ gc.geometryCollections.flatMap(keysFromGC)
-            keys
-          }
-
-          keysFromGC(gc)
-            .map(clipToKey(_, gcPredicates))
-            .flatten
-            .iterator
-      }
-
-    iterator
+    clipped.toOption
   }
+
+  /** Clips Features to a 3x3 grid surrounding the current Tile.
+    * This has been found to capture ''most'' Features which stretch
```

Comment:
Found to capture where? In VectorPipe? We could be using ClipToGrid in a setting where there are tiny tiles and big polygons, so I don't think this comment applies across the board.

byBufferedExtent is a pretty VP specific use case. Not sure this should be elevated into GeoTrellis, or just kept in VP.